packages/shaders/src/shader-mount.ts

```diff
@@ -165,76 +178,74 @@ export class ShaderMount {
     });
 
     this.resizeObserver.observe(this.parentElement);
-    visualViewport?.addEventListener('resize', this.handleVisualViewportChange);
-
-    const rect = this.parentElement.getBoundingClientRect();
-    this.parentWidth = rect.width;
-    this.parentHeight = rect.height;
-    this.handleResize();
   };
 
   // Visual viewport resize handler, mainly used to react to browser zoom changes.
-  // Wait 2 frames to align with when the resize observer callback is done (in case it might follow):
-  // - Frame 1: a paint after the visual viewport resize
-  // - Frame 2: a paint after the resize observer has been handled, if it was ever triggered
-  //
-  // Both resize observer and visual viewport will react to classic browser zoom changes,
-  // so we dedupe the callbacks, but pinch zoom only triggers the visual viewport handler.
-  private resizeRafId: number | null = null;
+  // Resize observer by itself does not react to pinch zoom, and although it usually
+  // reacts to classic browser zoom, it's not guaranteed in edge cases.
+  // Since timing between visual viewport changes and resize observer is complex
+  // and because we'd like to know the device pixel sizes of elements, we just restart
+  // the observer to get a guaranteed fresh callback regardless if it would have triggered or not.
   private handleVisualViewportChange = () => {
-    if (this.resizeRafId !== null) {
-      cancelAnimationFrame(this.resizeRafId);
-    }
+    this.resizeObserver?.disconnect();
+    this.setupResizeObserver();
 
-    this.resizeRafId = requestAnimationFrame(() => {
-      this.resizeRafId = requestAnimationFrame(() => {
-        this.handleResize();
-      });
-    });
+    // In case of debugging timing, from here on:
+    // - animation frame 1: a paint after the visual viewport resize
+    // - animation frame 2: a paint after the resize observer has been handled, if it was ever triggered
```

Comment:
This now reacts to resize via visual viewport change a frame faster

Reply:
hell yeah